src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -1212,45 +1212,165 @@ bool ViewProviderSketch::mouseButtonPressed(int Button, bool pressed, const SbVe
+
+                    if (SubNames.size() == 1) {
+                        noSelection = false;
+                        if (SubNames.begin()->substr(0, 4) == "Edge")
+                            selectLine = true;
+                        else if (SubNames.begin()->substr(0, 4) == "Vert")
+                            selectPoint = true;
+                        else if (SubNames.begin()->substr(0, 4) == "Cons")
+                            selectConstraint = true;
                     }
-                    else if (preselection.isPreselectCurveValid()) {
-                        return true;
+                    else if (SubNames.size() > 1) {
+                        noSelection = false;
+                        selectMultiple = true;
+                        int selectedLines = 0;
+                        int selectedPoints = 0;
+                        int selectedConstraints = 0;
+                        // go through the selected subelements
+                        for (std::vector<std::string>::const_iterator it = SubNames.begin();
+                            it != SubNames.end();
+                            ++it) {
+                            if (it->substr(0, 4) == "Edge") {
+                                ++selectedLines;
+                            }
+                            else if (it->substr(0, 4) == "Vert"){
+                                ++selectedPoints;
+                            }
+                            else if (it->substr(0, 4) == "Cons"){
+                                ++selectedConstraints;
+                            }
+                        }// End of Iteration
+                        if (selectedLines > 1 && selectedPoints == 0 && selectedConstraints == 0)
+                            selectLine = true;
+                        else if (selectedLines == 0 && selectedPoints > 1 && selectedConstraints == 0) {
+                            selectPoint = true;
+                        }
+                        else if (selectedLines == 0 && selectedPoints == 0 && selectedConstraints > 1){
+                            selectConstraint = true;
+                        }
+                        else if (selectedLines == 1 && selectedPoints > 0 && selectedConstraints == 0){
+                            selectLine = true;
+                            selectPoint = true;
+                        }
                     }
-                    else if (!preselection.PreselectConstraintSet.empty()) {
-                        return true;
                     }
+
+                    if (!noSelection){
+                        if (selectConstraint) {
+                        geom << "Sketcher_SelectElementsAssociatedWithConstraints"
+                             << "Sketcher_ToggleDrivingConstraint"
+                             << "Sketcher_ToggleActiveConstraint"
+                             << "Separator"
+                             << "Std_Delete";
+                             goto menucreation;
```

Comment:
goto is seldomly used. If you can do without it I think it is better as people are not very used to it.

Reply:
I need one goto if only constraints are selected, because the rest of the context menu makes no sense for constraints only but it works.